heaps/min_heap.py

```diff
@@ -57,18 +70,42 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(log n)
         """
-        pass
+        if index == 0:
+            return
+        if self.store[index].key < self.store[(index - 1) // 2].key:
```

Comment:
Notice that you repeated the `(index - 1) // 2` calculation a few times. Consider performing that once and storing it in a local variable.